src/python/pants/base/specs.py

```diff
@@ -41,7 +41,11 @@ class AddressLiteralSpec(Spec):
     def __str__(self) -> str:
```

Comment:
Since this isn't (very) perf sensitive, it might make sense to construct an `Address` and render it?

Reply:
The tricky thing is handling file targets, which need to have parsed out `relative_file_path` whereas `AddressLiteralSpec` keeps it as just `path_component`